drivers/system_pathways/app/models/system_pathways/report.rb

```diff
@@ -0,0 +1,519 @@
+            sort_by { |en| en.exit_date || Date.tomorrow }.
+            group_by(&:computed_project_type).
+            # keep only the last enrollment in each category
+            transform_values(&:last).
+            values
+
+          accepted_enrollments = accept_enrollments(involved_enrollments)
+          final_enrollment = accepted_enrollments.last
+          next unless final_enrollment.present?
+
+          returned_enrollment = enrollments.detect do |en|
+            # If we're still enrolled, we can't return
+            next false if final_enrollment.exit_date.blank?
+            # If our final enrollment didn't exit to a permanent destination we can't return
+            next false unless HudUtility.destination_type(final_enrollment.destination) == 'Permanent'
+            # If this enrollment starts before the end of the final enrollment, we can't have returned
+            next false unless en.entry_date > final_enrollment.exit_date
+
+            en.homeless?
+          end
+
+          # NOTE: we'll calculate age from the latter of the first enrollment entry date or report start
+          date = [accepted_enrollments.first.entry_date, filter.start].max
+          days_to_return = nil
+          days_to_return = returned_enrollment.entry_date - final_enrollment.exit_date if returned_enrollment.present?
+          report_client = report_clients[client] || Client.new(
+            client_id: client.id,
+            report_id: id,
+            first_name: client.first_name,
+            last_name: client.last_name,
+            personal_ids: client.source_clients.map(&:personal_id).uniq.join('; '),
+            dob: client.dob,
+            age: client.age(date),
+            am_ind_ak_native: client.am_ind_ak_native == 1,
+            asian: client.asian == 1,
+            black_af_american: client.black_af_american == 1,
+            native_hi_pacific: client.native_hi_pacific == 1,
+            white: client.white == 1,
+            ethnicity: client.ethnicity,
+            male: client.male == 1,
+            female: client.female == 1,
+            transgender: client.transgender == 1,
+            questioning: client.questioning == 1,
+            no_single_gender: client.no_single_gender == 1,
+            veteran_status: client.veteran_status,
+            involves_ce: served_by_ce,
+            destination: final_enrollment.destination,
+            destination_homeless: HudUtility.destination_type(final_enrollment.destination) == 'Homeless',
+            destination_temporary: HudUtility.destination_type(final_enrollment.destination) == 'Temporary',
+            destination_institutional: HudUtility.destination_type(final_enrollment.destination) == 'Institutional',
+            destination_other: HudUtility.destination_type(final_enrollment.destination) == 'Other',
+            destination_permanent: HudUtility.destination_type(final_enrollment.destination) == 'Permanent',
```

Comment:
These are mutually exclusive, maybe have a single field?

Reply:
Hmm, they are, this seemed easier to reason about.    I don't think there's any harm leaving them as is for now, but good to note.

Reply:
As long as the logic sets one, yes no harm.